pkg/sink/kafka/factory.go

```diff
@@ -17,257 +17,179 @@ import (
 	"context"
 	"time"
 
-	"github.com/IBM/sarama"
-	"github.com/pingcap/errors"
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
 	"github.com/pingcap/log"
 	commonType "github.com/pingcap/ticdc/pkg/common"
-	cerror "github.com/pingcap/ticdc/pkg/errors"
+	"github.com/pingcap/ticdc/pkg/errors"
 	"github.com/pingcap/ticdc/pkg/sink/codec/common"
 	"go.uber.org/zap"
 )
 
-// Factory is used to produce all kafka components.
-type Factory interface {
-	// AdminClient return a kafka cluster admin client
-	AdminClient() (ClusterAdminClient, error)
-	// SyncProducer creates a sync producer to writer message to kafka
-	SyncProducer() (SyncProducer, error)
-	// AsyncProducer creates an async producer to writer message to kafka
-	AsyncProducer(ctx context.Context) (AsyncProducer, error)
-	// MetricsCollector returns the kafka metrics collector
-	MetricsCollector(adminClient ClusterAdminClient) MetricsCollector
-}
-
-// FactoryCreator defines the type of factory creator.
-type FactoryCreator func(context.Context, *Options, commonType.ChangeFeedID) (Factory, error)
+const defaultTimeoutMs = 1000
 
-// SyncProducer is the kafka sync producer
-type SyncProducer interface {
-	// SendMessage produces a given message, and returns only when it either has
-	// succeeded or failed to produce. It will return the partition and the offset
-	// of the produced message, or an error if the message failed to produce.
-	SendMessage(ctx context.Context,
-		topic string, partitionNum int32,
-		message *common.Message) error
+type factory struct {
+	config       *kafka.ConfigMap
+	changefeedID commonType.ChangeFeedID
+	options      *Options
+}
 
-	// SendMessages produces a given set of messages, and returns only when all
-	// messages in the set have either succeeded or failed. Note that messages
-	// can succeed and fail individually; if some succeed and some fail,
-	// SendMessages will return an error.
-	SendMessages(ctx context.Context, topic string, partitionNum int32, message *common.Message) error
+// NewFactory returns a factory implemented based on kafka-go
+func NewFactory(
+	options *Options,
+	changefeedID commonType.ChangeFeedID,
+) (Factory, error) {
+	config := NewConfig(options)
+	return &factory{
+		config:       config,
+		changefeedID: changefeedID,
+		options:      options,
+	}, nil
+}
 
-	// Close shuts down the producer; you must call this function before a producer
-	// object passes out of scope, as it may otherwise leak memory.
-	// You must call this before calling Close on the underlying client.
-	Close()
+func (f *factory) AdminClient() (ClusterAdminClient, error) {
+	client, err := kafka.NewAdminClient(f.config)
+	if err != nil {
+		return nil, err
+	}
+	return newClusterAdminClient(client, f.changefeedID, defaultTimeoutMs), nil
```

Comment:
the `defaultTimeoutMs` is hard code to 1000, so simply set it inside the method `newClusterAdminClient`, no need to pass by argument.